shell/platform/embedder/embedder.h

```diff
@@ -552,6 +578,110 @@ typedef struct {
   const FlutterTaskRunnerDescription* platform_task_runner;
 } FlutterCustomTaskRunners;
 
+typedef struct {
+  FlutterOpenGLTargetType type;
+  union {
+    FlutterOpenGLTexture texture;
+    FlutterOpenGLFramebuffer framebuffer;
+  };
+} FlutterOpenGLCompositorLayerRenderer;
+
+typedef struct {
+  const void* allocation;
+  size_t row_bytes;
+  size_t height;
+  void* user_data;
+  VoidCallback destruction_callback;
+} FlutterSoftwareCompositorLayerRenderer;
+
+typedef enum {
+  kFlutterCompositorLayerRendererTypeOpenGL,
+  kFlutterCompositorLayerRendererTypeSoftware,
+} FlutterCompositorLayerRendererType;
+
+typedef struct {
+  // The size of this struct. Must be sizeof(FlutterCompositorLayerRenderer).
+  size_t struct_size;
+  // A baton that is not interpreted by the engine in any way. The embedder may
+  // use this to associate resources that are tied to the lifecycle of the
+  // |FlutterCompositorLayerRenderer|.
+  void* user_data;
+  FlutterCompositorLayerRendererType type;
+  union {
+    FlutterOpenGLCompositorLayerRenderer open_gl;
+    FlutterSoftwareCompositorLayerRenderer software;
+  };
+} FlutterCompositorLayerRenderer;
+
+typedef struct {
+  double x;
+  double y;
+} FlutterPoint;
```

Comment:
The inconsistency with existing APIs taking points and sizes is unfortunate. Have you thought about maybe doing a follow-up to add updated versions of the relevant APIs (window metrics, pointer)?

Reply:
I am not too excited about making that breaking API change (though the ABI would be stable). Maybe I should just place these inline in the struct instead of adding these additional structs for consistency?

Reply:
I was actually thinking you could freeze the old APIs, comment them as deprecated, and make new ones that used the new structs.

Reply:
We can do that too. Though, I don't think this patch needs to include that change. But if we agree that this is the direction we want to go, I'll keep these structs around.